manifests/server.pp

```diff
@@ -1,64 +1,218 @@
-# @param manage_gcc installs gcc
-# @param devices hash containing all the devices hyperglass can connect to. Defaults to demo data so the service starts properly.
-# @param commands specific commands that can be used by the devices
-# @param data generic hyperglass configuration hash.
+# @param manage_depended_services
+#   if true, installs all other services that hyperglass requires, like redis,
+#   yarn, nginx, python
+# @param manage_gcc
+#   installs gcc
+# @param manage_node
+#   Includes the nodejs class using version 14.x
+# @param manage_nginx
+#   Includes the nginx class
+# @param manage_python
+#   installs python3
+# @param manage_redis
+#   Installs redis5 from SCL.
+# @param manage_yarn
+#   Manages yarn repo and package.
+# @param manage_user
+#   When true, the user 'hyperglass' is managed.
+# @param manage_group
+#   When true, the group 'hyperglass' is managed.
+# @param python_version
+#   Specifies which version of python will be used by python::pyvenv if the
+#   `python3_version` fact is not found.
+# @param devices
+#   Hash containing all the devices managed by hyperglass.
+# @param commands
+#   Specific commands that can be used by the devices
+# @param data
+#   Generic hyperglass configuration hash.
 #
 # @see https://hyperglass.io/
 #
-# @author Tim Meusel <tim@bastelfreak.de>
 class hyperglass::server (
   Boolean $manage_depended_services = true,
-  Boolean $manage_python            = true,
-  Boolean $manage_gcc               = true,
-  Hash $data                        = {},
-  Hash $commands                    = {},
-  Hash $devices                     = {
-    'routers' => [
-      {
-        'name' => 'atl_router01',
-        'address' => '10.0.0.2',
-        'network' => {
-          'name' => 'secondary',
-          'display_name' => 'That Other Network',
-        },
-        'credential' => {
-          'username' => 'user2',
-          'password' => ' secret2',
-        },
-        'display_name' => 'Atlanta, GA',
-        'port' => 22,
-        'nos' => 'juniper',
-        'vrfs' => [
-          {
-            'name' => 'default',
-            'display_name' => 'Global',
-            'ipv4' => {
-              'source_address' => '192.0.2.2',
-            },
-          },
-        ],
-      },
-    ],
-  },
+  Boolean $manage_gcc = true,
+  Boolean $manage_node = true,
+  Boolean $manage_nginx = true,
+  Boolean $manage_python = true,
+  Boolean $manage_redis = true,
+  Boolean $manage_yarn = true,
+  Boolean $manage_user = true,
+  Boolean $manage_group = true,
+  String[1] $python_version = '3.6',
+  Hash $data = {},
+  Hash $commands = {},
+  Hash $devices = {},
 ) {
-  unless $facts['os']['name'] in ['CentOS', 'RedHat', 'VirtuozzoLinux'] {
-    fail('the hyperglass::server class currently only works on CentOS/RedHat')
+  if $facts['os']['family'] != 'RedHat' {
+    warning('This module only supports the RedHat family of platforms.')
   }
 
+  include hyperglass::agent
```

Comment:
why does the server now include the agent? They can be installed on different platforms and the agent isn't a requirement for the server.

Reply:
Why would you not run the agent on the server? I can separate it out, though it seems odd that you would install the agent all over the place and then say, but not on the lg server itself. It doesn't make sense from the Ops perspective.

Reply:
it makes sense for environments like mine, where hyperglass connects to real routers. The agent is simply not needed.

Reply:
also I think the agent can only talk to a local frr/bird instance not a remote one. so unless your hyperglass server acts as a router as well, the agent cannot be used.

Reply:
gotcha, i'll revert that

Reply:
Would be way easier if they had an rpm huh? :)

Reply:
Put common bits in hyperglass class and the server does not require the agent.